traitsui/testing/tests/test_ui_tester.py

```diff
@@ -0,0 +1,182 @@
+
+
+class SimpleApplication(ModelView):
+
+    model = Instance(Child)
+
+
+@requires_one_of([QT, WX])
+class TestUITesterIntegration(unittest.TestCase):
+    """ Integration tests for UITester.
+
+    These tests are close imitations for how UITester is expected to be used.
+    """
+
+    def setUp(self):
+        # UITest can also be used as a context manager
+        self.tester = UITester()
+        self.tester.start()
+
+    def tearDown(self):
+        self.tester.stop()
+
+    def test_instance_text_editor_from_view(self):
+        # Test popagating commands to instance simple/custom editors
+        father = Parent(first_name="M", last_name="C")
+        mother = Parent(first_name="J", last_name="E", age=50)
+        child = Child(mother=mother, father=father)
+        view = View(
+            Item("model.father", style="simple"),
+            Item("model.mother", style="custom"),
+        )
+        app = SimpleApplication(model=child)
+        with self.tester.create_ui(app, dict(view=view)) as ui:
+            self.tester.set_text(ui, "father.first_name", "B")
+            self.assertEqual(app.model.father.first_name, "B")
+
+            self.tester.set_text(ui, "mother.age", "A")  # invalid
+            self.assertEqual(app.model.mother.age, 50)
+
+            self.tester.set_text(ui, "mother.age", "60")  # valid
+            self.assertEqual(app.model.mother.age, 60)
+
+            self.tester.set_text(ui, "father.employment_status", "unemployed")
+            self.assertEqual(app.model.father.employment_status, "unemployed")
+
+    def test_instance_text_editor_query(self):
+        # Test popagating queries to instance simple/custom editors
+        father = Parent(first_name="M", last_name="C")
+        mother = Parent(first_name="J", last_name="E", age=50)
+        child = Child(mother=mother, father=father)
+        app = SimpleApplication(model=child)
+        view = View(
+            Item("model.father", style="simple"),
+            Item("model.mother", style="custom"),
+        )
+        with self.tester.create_ui(app, dict(view=view)) as ui:
+            app.model.father.first_name = "B"
+            actual = self.tester.get_text(ui, "father.first_name")
+            self.assertEqual(actual, "B")
+
+            app.model.mother.age = 60
+            actual = self.tester.get_text(ui, "mother.age")
+            self.assertEqual(actual, "60")
+
+            app.model.father.employment_status = "unemployed"
+            actual = self.tester.get_text(ui, "father.employment_status")
+            self.assertEqual(actual, "unemployed")
+
+
+# Test contributing custom simulator methods
+LOCAL_REGISTRY = SimulatorRegistry()
+
+if is_current_backend_qt4():
+
+    from traitsui.qt4.enum_editor import SimpleEditor as QtEnumEditor
+
+    @simulate(QtEnumEditor, LOCAL_REGISTRY)
+    class QtCustomSimulator(BaseSimulator):
+
+        def click_some_index(self, index):
+            self.editor.control.setCurrentIndex(index)
```

Comment:
Ideally this should be `self.editor.currentIndexChanged.emit(index)`, but the editor does not connect the handler for the integer implementation of `currentIndexChanged`, it connects to the string implementation instead. The better implementation for this simulator is to get the list of strings displayed *from* the Qt widget, and then get the text for the corresponding index, and then call `self.editor.control.currentIndexChanged(text)`...